### ISSUE TITLE
Add timestamps to messages

### ISSUE THREAD

**Issue by lazerwalker** [lazerwalker/azure-mud#80]
This is actually slightly difficult to think about design-wise. 

Slack/Discord both show timestamps as long as the previous message wasn't by the same user and was within X minutes of the previous message. They have a message design that gives them breathing room to do that, since the username is on a different line from the message. We don't have room to do that.

Other than design considerations, we don't currently store timestamps for messages. Two ways to think about this: either the server can inject authoritative timestamps for every message, or the local client can add a timestamp when it receives a message over the network. The latter is easier, but I suspect we will need the former for when we build out admin tooling. 

My plan for admin tooling is to wrap every SignalR message send in a wrapper function that also logs messages to disk. My instinct is that wrapper function can also modify each message by adding a timestamp onto the payload. So this is likely blocked on that.

I think this will eventually be a "good first issue", but holding off for now since it's entangled in a complex server change I'll likely make myself.

**Comment by lazerwalker** [comment]
Closed by mistake, PR #81 should have closed #70 ;)

**Comment by lazerwalker** [comment]
Design options:

1. Make messages look more like Discord/Slack default, with a dedicated line for username/timestamp
2. Put timestamps to the left, ignoring timestamps if the previous message was the same user and they are within X minutes
3. Horizontal line for "it's been a long time since the last message
4. Timestamps on the right

My instinct is 1 + 3.

**Comment by MoyTW** [comment]
If we do #1 + #3 we should also probably collapse consecutive same-member lines. Also we should vary the username/timestamp & spacing between the same-member lines, otherwise the username+timestamp could too easily be confused with the messages when scanning.

I personally prefer #2 because without avatars I find it hard to parse names, but with a good redesign I think 1+3 would work.

**Comment by lazerwalker** [comment]
Err, in my brain I meant 2+3, we're on the same page 🤣. I think 1 would look horrible with our aesthetic. 

Also agree about collapsing consecutive lines.

**Comment by lazerwalker** [comment]
For a naive first implementation, adding timestamps on the client-side based on when the message is received. We may eventually want to generate these on the server-side.

**Comment by lazerwalker** [comment]
This is janky, but in-place.

Leaving this open for a few days to feel it out. Things that might pop up:

- Reasons to move this to the server
- Figuring out the jankiness around when message.timestamp is a string vs a date
- CSS bugs
- Adding a horizontal line for "it's been a while"
- Changing the text format of our timestamps

**Comment by lazerwalker** [comment]
I think I'm mostly happy with how this works, other than fixing the scroll bugs that have popped up as a result of this. Created a separate issue for adding a horizontal line.
